app/helpers/marks_graders_helper.rb

```diff
@@ -1,86 +1,41 @@
 module MarksGradersHelper
-
-  # Given a list of graders and a grade entry form, constructs an array of
-  # table rows to be inserted into the graders FilterTable in the graders view.
-  # Called whenever it is necessary to update the graders table with multiple
-  # changes.
-  def construct_grader_table_rows(graders, grade_entry_form)
-    result = {}
-    graders.each do |grader|
-      result[grader.id] = construct_grader_table_row(grader, grade_entry_form)
+  # Return an array of TAs with the number of students they are assigned to on
+  # this marks graders form
+  def get_marks_graders_table_info(grade_entry_form)
+    graders = Ta.all
+
+    graders.map do |grader|
+      s = grader.attributes
+      s[:num_students] = grader.get_membership_count_by_grade_entry_form(
+          grade_entry_form)
+      s
     end
-
-    result
   end
 
-  # Given a list of students and a grade entry form, constructs an array of
-  # table rows to be inserted into the students FilterTable in the graders view.
-  # Called whenever it is necessary to update the students table with multiple
-  # changes.
-  def construct_table_rows(students, grade_entry_form)
-    result = {}
-    students.each do |student|
-      result[student.id] = construct_table_row(student, grade_entry_form)
-    end
-
-    result
-  end
-
-  def construct_table_row(student, grade_entry_form)
-    grade_entry_student = student.grade_entry_students.find do |entry|
-      entry.grade_entry_form_id == grade_entry_form.id
-    end
-
-      table_row = {}
-
-      table_row[:id] = student.id
-      table_row[:filter_table_row_contents] =
-        render_to_string(
-          partial: 'marks_graders/table_row/filter_table_row',
-          formats: [:html], handlers: [:erb],
-          locals: { student: student, grade_entry_student: grade_entry_student }
-        )
-
-      #These are used for sorting
-      table_row[:user_name] = student.user_name
-      table_row[:first_name] = student.first_name
-      table_row[:last_name] = student.last_name
-      table_row[:section] = student.section.nil? ? '' : student.section.name
-      table_row[:members] =
-        if grade_entry_student
-          grade_entry_student.grade_entry_student_tas.map do |gest|
-            gest.ta.user_name
-          end.join(',')
-        else
-          ''
+  # Return an array of the students for this form and their assigned TAs
+  def get_marks_graders_student_table_info(students, grade_entry_form)
```

Comment:
Assignment Branch Condition size for get_marks_graders_student_table_info is too high. [17.29/15]